### PR TITLE
Make show and close of iOS views respect Animated

### PR DESCRIPTION
**Pull request by martijn00**
### :sparkles: What kind of change does this PR introduce? (Bug fix, feature, docs update...)

Bug fix

### :arrow_heading_down: What is the current behavior?

Animated is not respected.

### :new: What is the new behavior (if this is a feature change)?

Attributes are passed to the method so it can be used.

### :boom: Does this PR introduce a breaking change?

Yes

### :memo: Links to relevant issues/docs

Fixes https://github.com/MvvmCross/MvvmCross/issues/2762

### :thinking: Checklist before submitting

- [ ] All projects build
- [ ] Follows style guide lines ([code style guide](https://github.com/MvvmCross/MvvmCross#code-style-guidelines))
- [ ] Relevant documentation was updated ([docs style guide](https://www.mvvmcross.com/documentation/contribute/mvvmcross-docs-style-guide))
- [ ] Rebased onto current develop

### CODE REVIEW

File: MvvmCross/Platforms/Ios/Presenters/Attributes/MvxDetailSplitViewPresentationAttribute.cs [hunk 1]
```diff
@@ -1,14 +0,0 @@
-// Licensed to the .NET Foundation under one or more agreements.
-// The .NET Foundation licenses this file to you under the MS-PL license.
-// See the LICENSE file in the project root for more information.
-
-using MvvmCross.Presenters.Attributes;
-
-namespace MvvmCross.Platforms.Ios.Presenters.Attributes
-{
-    public class MvxDetailSplitViewPresentationAttribute : MvxBasePresentationAttribute
```
Comment (nmilcoff) [inline MvvmCross/Platforms/Ios/Presenters/Attributes/MvxDetailSplitViewPresentationAttribute.cs]: Although I think this change makes sense, there's no information about this on the PR description. Please add it, it makes my life so much easier when writing migration docs. Also improves traceability

File: MvvmCross/Platforms/Ios/Presenters/MvxIosViewPresenter.cs [hunk 2]
```diff
@@ -370,28 +363,23 @@ protected virtual bool CloseTabViewController(IMvxViewModel viewModel, MvxTabPre
...
-            return CloseModalViewController(viewModel);
-        }
-
-        protected virtual bool CloseMasterSplitViewController(IMvxViewModel viewModel, MvxMasterSplitViewPresentationAttribute attribute)
-        {
-            if (SplitViewController != null && SplitViewController.CloseChildViewModel(viewModel))
+            if (SplitViewController != null && SplitViewController.CloseChildViewModel(viewModel, attribute))
                 return true;
 
             return true;
```
Comment (nmilcoff) [inline MvvmCross/Platforms/Ios/Presenters/MvxIosViewPresenter.cs]: This line should return false, shouldn't it?

File: MvvmCross/Platforms/Ios/Presenters/MvxIosViewPresenter.cs [hunk 3]
```diff
@@ -370,28 +363,23 @@ protected virtual bool CloseTabViewController(IMvxViewModel viewModel, MvxTabPre
...
         }
 
-        protected virtual bool CloseDetailSplitViewController(IMvxViewModel viewModel, MvxDetailSplitViewPresentationAttribute attribute)
+        protected virtual bool CloseDetailSplitViewController(IMvxViewModel viewModel, MvxSplitViewPresentationAttribute attribute)
         {
-            if (SplitViewController != null && SplitViewController.CloseChildViewModel(viewModel))
+            if (SplitViewController != null && SplitViewController.CloseChildViewModel(viewModel, attribute))
                 return true;
 
             return true;
```
Comment (nmilcoff) [inline MvvmCross/Platforms/Ios/Presenters/MvxIosViewPresenter.cs]: This line should return false, shouldn't it?

File: MvvmCross/Platforms/Ios/Presenters/MvxIosViewPresenter.cs [hunk 4]
```diff
@@ -370,28 +363,23 @@ protected virtual bool CloseTabViewController(IMvxViewModel viewModel, MvxTabPre
...
         }
 
-        protected virtual bool CloseModalViewController(IMvxViewModel toClose)
+        protected virtual bool CloseModalViewController(IMvxViewModel toClose, MvxModalPresentationAttribute attribute)
```
Comment (nmilcoff) [inline MvvmCross/Platforms/Ios/Presenters/MvxIosViewPresenter.cs]: Many users will hit this breaking change, just leaving a note as a reminder for docs

**Comment by martijn00** [comment]
@nmilcoff i've updated the docs

**Comment by MagneticLlama** [comment]
Is this still a known problem? I've been trying to disable view animations in iOS, and I want to make sure it's not something on my end.

**Comment by Cheesebaron** [comment]
@MagneticLlama please create a new issue rather than commenting on Pull Requests merged 2 years ago. Thanks!

**Pull request merged**